server/routes/api/v1/auth.js

```diff
@@ -0,0 +1,155 @@
+const express = require('express');
+const router = express.Router();
+const auth = require('../../../middleware/auth');
+const bcrypt = require('bcryptjs');
+const config = require('config');
+const { check, validationResult } = require('express-validator');
+const jwt = require('jsonwebtoken');
+const db = require('./dbinsert');
+const gravatar = require('gravatar');
+
+const cors = require('cors');
+router.use(cors());
+
+// @route GET api/auth
+// @ desc returns the user
+// @access Public
+
+router.get('/', auth, async (req, res) => {
+  try {
+    const user = await db.querydb(req.user.id);
+    res.json({ user });
+  } catch (err) {
+    console.error(err.message);
+    res.status(500).send('Server error');
```

Comment:
One very important to bear in mind when building an API is consistency. Above you returned a JSON but here you are returning text. You should return JSON throughout your API